File: frontend/src/types/matchEvents.ts
import type { Player } from './player';
import type { Team } from './team';

type EventType =
  | 'goal'
  | 'own_goal'
  | 'penalty_goal'
  | 'penalty_miss'
  | 'yellow_card'
  | 'red_card'
  | 'substitution'
  | 'var'
  | 'other';

export interface MatchEvent {
  team: Team;
  player?: Player;
  secondary_player?: Player;
  player_name?: string;
  secondary_player_name?: string;
  player_external_id?: number;
  secondary_player_external_id?: number;
  event_type: EventType;
  minute: number;
  extra_minute?: number;
  detail?: string;
  comments?: string;
}

File: backend/app/schemas/match_events.py
from pydantic import BaseModel, ConfigDict

from app.models.match_event import EventType
from app.schemas.common import TeamSummary
from app.schemas.metadata import ResponseMetadata
from app.schemas.players import PlayerSummary


class MatchEventItemResponse(BaseModel):
    team: TeamSummary

    player: PlayerSummary | None = None
    secondary_player: PlayerSummary | None = None

    player_name: str | None = None
    secondary_player_name: str | None = None

    player_external_id: int | None = None
    secondary_player_external_id: int | None = None

    event_type: EventType

    minute: int
    extra_minute: int | None = None

    detail: str | None = None
    comments: str | None = None

    model_config = ConfigDict(from_attributes=True)


class MatchEventsResponse(BaseModel):
    data: list[MatchEventItemResponse]
    metadata: ResponseMetadata


class MatchEventRefreshRow(BaseModel):
    external_match_id: int
    external_team_id: int

    player_external_id: int | None = None
    secondary_player_external_id: int | None = None

    player_name: str | None = None
    secondary_player_name: str | None = None

    event_type: EventType

    minute: int
    extra_minute: int | None = None

    detail: str | None = None
    comments: str | None = None
